lib/core/src/display/navigation/navigator.rs

```diff
@@ -0,0 +1,158 @@
+use crate::prelude::*;
+
+use super::EventHandler;
+use super::Event;
+use super::ZoomEvent;
+use super::PanEvent;
+use crate::system::web::EventListeningResult as Result;
+use crate::display::rendering::Camera;
+use crate::display::rendering::CameraType;
+use crate::display::rendering::DOMContainer;
+
+use nalgebra::Vector3;
+
+// =================
+// === Animation ===
+// =================
+
+/// Struct used to store animation related properties.
+struct Animation {
+    desired_pos  : Vector3<f32>,
+    vel          : Vector3<f32>,
+    desired_rate : f32,
+    time_count   : f32,
+    drag         : f32,
+    spring_coeff : f32,
+    mass         : f32,
+    min_vel      : f32
+}
+
+// =================
+// === Navigator ===
+// =================
+
+/// Navigator enables camera navigation with mouse interactions on the specified
+/// DOM.
+pub struct Navigator {
+    dom    : DOMContainer,
```

Comment:
Why camera navigator knows anything about a DOM ? This should not be true.

Reply:
The reason it knows about DOM is because we can target the mouse interactions to it. This enables us to navigate on two different viewports.

Reply:
Let me rephrase it.

The Navigator struct adds mouse event listeners to the specified DOM. This enables us to have two different DOMs, displaying two different scenes and we can navigate through them independently.

Is it clearer now?